daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -947,6 +947,12 @@ func (devices *DeviceSet) initDevmapper(doInit bool) error {
 		return graphdriver.ErrNotSupported
 	}
 
+	// https://github.com/docker/docker/issues/4036
+	if supported := devicemapper.UdevSetSyncSupport(true); !supported {
+		log.Warnf("Warning, Udev sync is not supported. Behavior may be unpredictable")
```

Comment:
@vbatts I'd change `Warning, Udev sync is not supported. Behavior may be unpredictable` to something along the lines of `WARNING: Udev sync is not supported. This will lead to unexpected behavior, data loss and errors`.
Perhaps we should also print the kind of errors (like the ones seen on #4036) one would see when printing this warning? This should make it easier to set expectations, rather than letting users ignore it.